src/Raven.Server/Documents/Sharding/Handlers/Processors/Documents/ShardedDocumentHandlerProcessorForGet.cs

```diff
@@ -142,11 +142,11 @@ protected override async ValueTask<DocumentsResult> GetDocumentsImplAsync(Transa
         IAsyncEnumerable<Document> documents;
         if (startsWith != null)
         {
-            documents = RequestHandler.DatabaseContext.Streaming.GetDocumentsAsyncById(streams, token);
+            documents = RequestHandler.DatabaseContext.Streaming.GetDocumentsAsyncById(streams, token).Select(x => x.Item);
         }
         else
         {
-            documents = RequestHandler.DatabaseContext.Streaming.GetDocumentsAsync(streams, token);
+            documents = RequestHandler.DatabaseContext.Streaming.GetDocumentsAsync(streams, token).Select(x => x.Item);
```

Comment:
Aren't those used in high traffic location? Wouldn't that be expensive via linq?

Reply:
it is `IAsyncEnumerable` so I think it shouldn't be expensive..

Reply:
My concern is on the `Select`, which does delegate allocation  & invocation

Reply:
done